src/toil/leader.py

```diff
@@ -572,22 +572,24 @@ def getNumberOfJobsIssued(self, preemptable=None):
           If true, return just the number of preemptable jobs. If false, return
           just the number of non-preemptable jobs.
         """
-        #assert self.jobsIssued >= 0 and self._preemptableJobsIssued >= 0
         if preemptable is None:
             return len(self.jobBatchSystemIDToIssuedJob)
         elif preemptable:
             return self.preemptableJobsIssued
         else:
             assert len(self.jobBatchSystemIDToIssuedJob) >= self.preemptableJobsIssued
             return len(self.jobBatchSystemIDToIssuedJob) - self.preemptableJobsIssued
+                                    
 
-    def getNumberAndAvgRuntimeOfCurrentlyRunningJobs(self):
+    def getNumberAndAvgRuntimeOfCurrentlyRunningJobs(self, nodeShape=None):
         """
         Returns a tuple (x, y) where x is number of currently running jobs and y
         is the average number of seconds (as a float)
         the jobs have been running for.
         """
         runningJobs = self.batchSystem.getRunningBatchJobIDs()
+        runningJobs = {jobID:runningJobs[jobID] for jobID in runningJobs if getNodeShapeForJob(jobShape=self.jobBatchSystemIDToJobNode[jobID], nodeShapes=self.provisioner.nodeShapes) == nodeShape}
```

Comment:
Similarly, the jobBatchSystemIDToJobNode attribute doesn't appear to be set anywhere either. Actually, it looks like this entire function is dead code, which is why these undefined variables don't cause an error. Can we remove this? (and, potentially, the version in clusterScalerTest?)